client/state/domains/suggestions/actions.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	DOMAINS_SUGGESTIONS_RECEIVE,
+	DOMAINS_SUGGESTIONS_REQUEST,
+	DOMAINS_SUGGESTIONS_REQUEST_FAILURE,
+	DOMAINS_SUGGESTIONS_REQUEST_SUCCESS
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that a domains suggestion object
+ * has been received.
+ *
+ * @param   {Array}    suggestions              domain suggestions
+ * @param   {Object}   queryObject              domain suggestions queryObject
+ * @returns {Object}   Action object
+ */
+export function receiveDomainsSuggestions( suggestions, queryObject ) {
+	return {
+		type: DOMAINS_SUGGESTIONS_RECEIVE,
+		queryObject,
+		suggestions
```

Comment:
Yes, we can consider now assemblers legacy approach. Once we introduced schema validation for persistence layer and selectors as a "proxy" for accessing states, assembler should be considered redundant. We should drop them :) It was introduced by me in the Flux era, and at that point it was the simplest way to pick needed data from server responses and make the data adhere to our coding guidelines. It gave was also some possibility to do simple validation and data types mapping. All data can be achieved now with aforementioned JSON schema and selectors.

In addition I think it's way much better idea to store unprocessed data inside Redux state, because of our persistence layer and server side rendering which needs to share state!